src/wfcopy.c

```diff
@@ -1797,26 +1797,37 @@ GetNextPair(PCOPYROOT pcr, LPTSTR pFrom,
 
    if (dwOp == OPER_MKDIR) {
 
-      //
-      // Make sure the new directory is not a subdir of the original...
-      // Assumes case insensitivity.
-      //
-      pT = pToPath;
+      if (!_wcsicmp(pFrom, pToPath)) {
+         switch (dwFunc) {
+         case FUNC_COPY:
+            lstrcat(pToPath, L" - Copy");
+            lstrcat(pcr->szDest, L" - Copy");
+            break;
+         default:
+            break;

```

Comment:
Can you explain when this would happen?

If I'm reading this section correctly, the previous code would return an error if the two paths were identical.  In these changes, FUNC_COPY appends rather than failing; but for anything except FUNC_COPY, identical strings hit this break and convert what was previously failure into success.  Is this correct?

Reply:
Almost correct. ;-)
The previous code checks if it was a *substring*, so it was not a strcmp(). See the original comment

`// The two fully qualified strings are equal up to the end of the`
`//   source directory ==> the destination is a subdir.Must return`
`//   an error.`

The new one checks if the path is *exactly* the same and then appends. 
If it is not exactly the same then we go the original way.

It does come here not with something other than FUNC_COPY. This is avoided by wfdirsrc.c:1055
Anyhow give it a try. It works:-)

Reply:
I saw the comment, but I was looking at the code:

```
if (!*pFrom && (!*pT || *pT == CHAR_BACKSLASH)) {
```

If the end of pFrom has been reached, and either the end of pT was reached or a backslash was reached.  So if I'm reading it right, it's checking for equality, or checking that pT contains additional path components.

Reply:
Are you saying that if pFrom and pT are the exact same value, the previous code would not have returned an error?

The way I'm reading it, the "!*pT" part of the check is for exactness; the "*pT == CHAR_BACKSLASH" part is checking for pT having extra components that pFrom does not have.  What would prevent this from returning failure on exact match?

If the previous code would have returned an error on exact match, this change appears to modify that.  FUNC_COPY is handled successfully, by adding a suffix.  Everything except FUNC_COPY hits "default: break;" and succeeds where it previously failed.

Reply:
Correct the method also check for exact matches, but also for prefixes. But this is a sidetrack.

It took me a little while to understand your concern but I see your point that in case of non FUNC_COPY the default: clause will succeed without error. So I put the error handling in now.
But this part of the code will never be reached. I put a breakpoint onto it, and played, but was not able to reach this. Anyhow. Better be safe and raise an error if it somehow comes around there.